### ISSUE TITLE
Remove compaction final state from code and in upgrade step

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#3465]
After #3460 the compaction final state metadata  was no longer needed.  This can be removed from the metadata schema and ample.  Also an upgrade step should be added that removes it from the metadata table.

Below is some of the code that needs to be removed.

https://github.com/apache/accumulo/blob/447b2e5dbaf5d59a00433d6bfc6ce8b49fd79f97/core/src/main/java/org/apache/accumulo/core/metadata/schema/MetadataSchema.java#L418

https://github.com/apache/accumulo/blob/447b2e5dbaf5d59a00433d6bfc6ce8b49fd79f97/core/src/main/java/org/apache/accumulo/core/metadata/schema/Ample.java#L205-L215
